### PR TITLE
Changelog update

### PR DESCRIPTION
**Pull request by rflperry**
#### What does this implement/fix? Explain your changes.
Drafted changelog updates reflecting current and ongoing changes for the next release. To be finalized and reviewed just prior to next version release.

#### Any other comments?
Introduces sklearn changelog [labels](https://scikit-learn.org/stable/whats_new/v0.23.html) and and styles (referencing PRs and authors).

### CODE REVIEW

File: docs/changelog.rst [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+- |API Change| : you will need to change your code to have the same effect in the future; or a feature will be removed in the future.
+
+Version 0.4.0
+-------------
+Date TBD. Updates in this release:
+
+`mvlearn.decomposition <https://github.com/mvlearn/mvlearn/tree/master/mvlearn/decomposition>`_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- |Feature| Adds GroupICA and GroupPCA. `#225 <https://github.com/mvlearn/mvlearn/pull/225>`_ by `Pierre Ablin <https://github.com/pierreablin>`_.
```
Comment (pierreablin) [inline docs/changelog.rst:24]:
```suggestion
- |Feature| Adds GroupICA and GroupPCA. `#225 <https://github.com/mvlearn/mvlearn/pull/225>`_ by `Pierre Ablin <https://github.com/pierreablin>`_ and `Hugo Richard <https://github.com/hugorichard>`_
```

**Review by pierreablin — APPROVED** [review]
Looks good :)

**Comment by rflperry** [comment]
Added in #225 , #228 , #229
